cmd/server/api/router.go

```diff
@@ -0,0 +1,17 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+type Router interface {
+	HandleFunc(pattern string, handler func(http.ResponseWriter, *http.Request)) *mux.Route
+	ServeHTTP(w http.ResponseWriter, r *http.Request)
+}
+
+func NewRouter(h ApiHandlerInterface, r Router) Router {
```

Comment:
I agree, maybe naming it `Router`, `Routes` or `RegisteredRoutes` would be more suitable. However, I would refrain from naming it 'SetRoutes' since the function itself doesn't actually set any routes; it merely manages routes associated with a function.